internal/cli/opsmanager/clusters/delete.go

```diff
@@ -43,44 +44,147 @@ func (opts *DeleteOpts) Run() error {
 	if !opts.Confirm {
 		return nil
 	}
+
+	hostIds, err := opts.newHostIds()
+	if err != nil {
+		return err
+	}
+
+	// shutdown cluster
+	err = opts.shutdownCluster()
+	if err != nil {
+		return err
+	}
+
+	// Remove cluster from automation
+	err = opts.removeClusterFromAutomation()
+	if err != nil {
+		return err
+	}
+
+	// Stop monitoring
+	err = opts.stopMonitoring(hostIds)
+	if err != nil {
+		return err
+	}
+
+	fmt.Print("Cluster deleted\n")
+	return nil
+}
+
+func (opts *DeleteOpts) newHostIds() ([]string, error) {
 	current, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+	if err != nil {
+		return nil, err
+	}
+
+	hostnameMap := make(map[string][]int32)
+	for _, process := range current.Processes {
+		if process.Args26.Replication.ReplSetName == opts.Entry {
+			hostnameMap[process.Hostname] = append(hostnameMap[process.Hostname], int32(process.Args26.NET.Port))
+		}
```

Comment:
[q] does this work for a sharded cluster as well?

Reply:
I am not sure, I have to test this scenario

Reply:
No it doesn’t. I see that we have `process.cluster` for `mongos` but we don't for `mongod`. `mongods` have 

```
replication": {"replSetName": "myShard_0"},
```

We may ask to the user to provide the shard name and use `contains("shardName")` here

Reply:
`process` has a `Cluster` prop, it's what I used when grouping processes of a sharded cluster but I don't remember if that's set for a replica set as well

Reply:
I am wondering what is the workflow with a shared cluster: should I shutdown the mongod first? configRS? or mongos? I am going to have a look to the API call that is used in the atlas ui.

Reply:
I'm testing this and it looks like the shutdown has a bug that us not sutting down the config RS nor the mongos, see https://github.com/mongodb/go-client-mongodb-ops-manager/blob/3bd88d15acfe2afed2bd8293c6457acfd99e3c2d/atmcfg/atmcfg.go#L29-L36

So we may need to fix that first

Then we have the issue of getting host ids, and maybe we should do something like the shutdown method and just walk thru all the arrays collecting what matches